### ISSUE TITLE
Issue with deepExtend and Arrays...

### ISSUE THREAD

**Issue by simonwidjaja** [HubSpot/youmightnotneedjquery#181]
I ran in a problem when using deepExtend() on an object that had a property 'children' of type Array.
Instead getting a extended new Object with children of type Array I got a children property of type Object.

This is my fix that worked for me:

```
      function deepExtend (out) {
        out = out || {};
        for (var i = 1; i < arguments.length; i++) {
          var obj = arguments[i];
          if (!obj)
            continue;
          for (var key in obj) {
            if (obj.hasOwnProperty(key)) {
              if (obj[key] instanceof Array)
                out[key] = deepExtend([], obj[key]);
              if (typeof obj[key] === 'object')
                out[key] = deepExtend(out[key], obj[key]);
              else
                out[key] = obj[key];
            }
          }
        }
        return out;
      };
```

**Comment by asimen1** [comment]
Also reported here #157

**Comment by Abbondanzo** [comment]
Fixed in #211
